FirebaseAdmin/FirebaseAdmin/Auth/Hash/Scrypt.cs

```diff
@@ -130,21 +62,42 @@ public int MemoryCost
         /// Returns the options for the hashing algorithm.
         /// </summary>
         /// <returns>
-        /// Dictionary defining options such as signer key, .
+        /// Dictionary defining options such as signer key.
         /// </returns>
-        protected override IReadOnlyDictionary<string, object> GetOptions()
+        protected override IReadOnlyDictionary<string, object> GetHashConfiguration()
         {
-            var dict = new Dictionary<string, object>((Dictionary<string, object>)base.GetOptions());
+            if (this.Key == null)
+            {
+                throw new ArgumentNullException("key must be initialized");
+            }
+
+            if (this.Key.Length == 0)
+            {
+                throw new ArgumentException("key must not be null or empty");
+            }
+
+            if (this.SaltSeparator == null)
```

Comment:
Make sure to base64 encode the byte arrays.

Reply:
For visibility after our discussion, Newtonsoft by default will Base64 encode byte arrays as shown [here](https://www.newtonsoft.com/json/help/html/SerializationGuide.htm#PrimitiveTypes)